File: src/main/java/lang/java/core/MethodOverriding.java
package lang.java.core;

public class MethodOverriding {
  public static void main(String[] args) {
    class A {
      public void print() {
        System.out.println("A");
      }
    }
    class B extends A {
      public void print() {
        super.print();
        System.out.println("B");
      }
    }
    new B().print();
  }
}

File: src/main/scala/lang/scala/core/MethodOverriding.scala
package lang.scala.core

object MethodOverriding extends App {
  class A {
    def print(): Unit = println("A")
  }
  class B extends A {
    override def print(): Unit = {
      super.print()
      println("B")
    }
  }
  new B().print()
}
